### PR TITLE
CI: migrate pipeline into GH Actions

### PR DESCRIPTION
**Pull request by jportasa**
- [x] Add version print logic into codebase 
- [x] Add docker login for pull rate limitations 
- [x] Add snyk step as `needs` for prerelease once it will be ok
- [x] Migrate into Org secrets
- [x] Change upgrade codes for windows (keep old codes)
- [x] Squash commits because Joan Porta did not accept CLA

### CODE REVIEW

File: .github/workflows/repolinter.yml [hunk 1]
```diff
@@ -1,31 +0,0 @@
...
-# Currently there is no elegant way to specify the default
-# branch in the event filtering, so branches are instead
-# filtered in the "Test Default Branch" step.
-on: [push, workflow_dispatch]
-
-jobs:
-  repolint:
-    name: Run Repolinter
-    runs-on: ubuntu-latest
-    steps:
```
Comment (paologallinaharbur) [inline .github/workflows/repolinter.yml:15]: Why this action was removed?

Should it be reintroduced?

File: build/package/windows/nri-amd64-installer/Product.wxs [hunk 2]
```diff
@@ -67,29 +67,33 @@
...
+                <File Id="FILE_NRI_$(var.IntegrationName)_DEFINITION_YML"
+                      Name="$(var.IntegrationName)-win-definition.yml"
+                      Source="$(var.BinariesPath)New Relic\newrelic-infra\newrelic-integrations\$(var.IntegrationName)-win-definition.yml"
                       KeyPath="yes"/>
             </Component>
         </ComponentGroup>
         <ComponentGroup Id="PluginConfigsComponent" Directory="PluginConfigsFolder">
-            <Component Id="CMP_NRI_{IntegrationName}_CONFIG_YML" Guid="93226d36-a575-4729-a273-87cc231bed0d" Win64="yes"
+            <Component Id="CMP_NRI_$(var.IntegrationName)_CONFIG_YML" Guid="93226d36-a575-4729-a273-87cc231bed0d" Win64="yes"
                        NeverOverwrite="yes" Permanent="yes">
```
Comment (paologallinaharbur) [inline build/package/windows/nri-amd64-installer/Product.wxs:91]: We should remove Permanent and Never Overwrite

File: build/package/windows/nri-386-installer/Product.wxs [hunk 3]
```diff
@@ -67,29 +67,33 @@
...
+                <File Id="FILE_NRI_$(var.IntegrationName)_DEFINITION_YML"
+                      Name="$(var.IntegrationName)-win-definition.yml"
+                      Source="$(var.BinariesPath)New Relic\newrelic-infra\newrelic-integrations\$(var.IntegrationName)-win-definition.yml"
                       KeyPath="yes"/>
             </Component>
         </ComponentGroup>
         <ComponentGroup Id="PluginConfigsComponent" Directory="PluginConfigsFolder">
-            <Component Id="CMP_NRI_{IntegrationName}_CONFIG_YML" Guid="93226d36-a575-4729-a273-87cc231bed0d" Win64="no"
+            <Component Id="CMP_NRI_$(var.IntegrationName)_CONFIG_YML" Guid="93226d36-a575-4729-a273-87cc231bed0d" Win64="no"
                        NeverOverwrite="yes" Permanent="yes">
```
Comment (paologallinaharbur) [inline build/package/windows/nri-386-installer/Product.wxs:91]: same here

File: .github/workflows/repolinter.yml (hunk 1 above)
Comment (roobre) [reply .github/workflows/repolinter.yml]: Seems like it was deleted by mistake at some point, I've brought it back.

File: build/package/windows/nri-amd64-installer/Product.wxs (hunk 2 above)
Comment (roobre) [reply build/package/windows/nri-amd64-installer/Product.wxs]: Flags have been obliterated.

File: build/package/windows/nri-386-installer/Product.wxs (hunk 3 above)
Comment (roobre) [reply build/package/windows/nri-386-installer/Product.wxs]: Same

**Comment by roobre** [comment]
Rebased to include #51

**Review by paologallinaharbur — APPROVED** [review]

**Pull request merged**